skorch/dataset.py

```diff
@@ -86,12 +82,10 @@ def __init__(
             self,
             X,
             y=None,
-            device='cpu',
```

Comment:
Could you please leave this here for now and down below raise a deprecation warning that "device is no longer needed by Dataset and will be ignored." with a comment to remove the warning in a later release.